core/src/main/java/feast/core/job/JobUpdateTask.java

```diff
@@ -73,147 +74,131 @@ public JobUpdateTask(
     this.currentJob = currentJob;
     this.jobManager = jobManager;
     this.jobUpdateTimeoutSeconds = jobUpdateTimeoutSeconds;
+    this.runnerType = jobManager.getRunnerType().toString();
   }
 
   @Override
   public Job call() {
     ExecutorService executorService = Executors.newSingleThreadExecutor();
```

Comment:
@zhilingc I'm trying to understand why this is used—`JobUpdateTask` is a `Callable` and its instances are all dispatched on separate threads by `JobCoordinatorService`, it seems redundant to have the tasks then use another thread again where they invoke a job manager (that will [usually] do network I/O). If anything, it would seem that futures should be at the layer of the job manager instead. `call()` submits and then blocks on them before returning.

One more question for you is about the parameters of helper methods like `startJob` and `updateJob`: most of them except the job itself are instance state of the `JobUpdateTask` itself, and they were able to be `final`, so it seems like there's no need for passing them around through arguments. I removed them in the process of working out what `call()` actually needed to touch. Am I missing anything, or were they just a vestige of some refactoring in your initial work?

Reply:
After reading through the code I'm not sure what my past self was thinking tbqh :( You're right, its completely redundant. 

As for the second point, you're right. I must've missed it.